lib/fs/filesystem.go

```diff
@@ -170,7 +170,8 @@ func NewFilesystem(fsType FilesystemType, uri string) Filesystem {
 	if l.ShouldDebug("filesystem") {
 		fs = &logFilesystem{fs}
 	}
-	return fs
+
+	return NewWalkFilesystem(fs)
```

Comment:
This will have an undesired side effect of logging tons of stuff in logFs, yet not a killer.

Reply:
I agree it is very noisy and I will happily undo that change, I just wanted to bring it up for discussion. It took me a while to figure out that nothing is working because `fs.Walk` isn't ordered, and I first had to figure out that the walkfs doesn't produce any debug logging to get there.